### ISSUE TITLE
Auto-scrolling on keyboard nav doesn't work on /r/NFL

### ISSUE THREAD

**Issue by leviroth** [honestbleeps/Reddit-Enhancement-Suite#4420]
- My browser (copy from https://www.whatismybrowser.com): Chrome 60
- My RES version: 5.8.4
- Am I in reddit beta (yes/no): no

*What's the problem?*

The auto-scrolling that should occur when I keyboard-navigate to an element that's off the visible page does not occur. Open any large comment thread on the front page and try to scroll between top-level comments with Ctrl-j or Ctrl-k. The page doesn't scroll. Tested with both legacy and "middle" scrolling option.

I tried looking at this in the debugger and it seemed to have something to do with the viewport being calculated as having a very large height (in the ballpark of 25000). However, I'm not familiar enough with Chrome development or the RES codebase to track it down further.

*Are there any related posts in https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues or /r/RESissues?*

Not that I could find.

*What other browser extensions are installed?*

I confirmed this bug in an incognito window with only RES.

**Comment by leviroth** [comment]
Looking into this a little more, it seems to be related to an inaccurate `document.documentElement.getBoundingClientRect()` height value. This is meant to determine the size and position of the current viewport, and it's used that way in the RES scrolling code, [e.g.](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/a4e78b8d1bda5d7a2842f327dac456c4e109ae5c/lib/utils/dom.js#L64). However, on /r/NFL's pages I get much larger values:

```
> document.documentElement.getBoundingClientRect()
DOMRect {x: 0, y: -1730, width: 1287, height: 45978, top: -1730, …}
bottom: 44248
height: 45978
left: 0
right: 1287
top: -1730
width: 1287
x: 0
y: -1730
__proto__: DOMRect
```

Perhaps a fix would be as simple as changing `getViewportSize` to use `document.documentElement.clientHeight` etc. instead of `getBoundingClientRect` - but I'm very far from a JS expert.
